src/wrappers/themis/Obj-C/objcthemis/smessage.m

```diff
@@ -62,72 +64,74 @@ - (nullable instancetype)initInSignVerifyModeWithPrivateKey:(NSData *)privateKey
 }
 
 
-- (nullable NSData *)wrapData:(nullable NSData *)message error:(NSError * __autoreleasing *)error {
+- (nullable NSData *)wrapData:(nullable NSData *)message error:(NSError *__autoreleasing *)error {
     size_t wrappedMessageLength = 0;
     TSErrorType result = TSErrorTypeFail;
 
     switch (self.mode) {
         case TSMessageModeEncryptDecrypt:
-            result = (TSErrorType) themis_secure_message_wrap([self.privateKey bytes], [self.privateKey length],
-                [self.publicKey bytes], [self.publicKey length], [message bytes], [message length],
-                NULL, &wrappedMessageLength);
+            result = (TSErrorType) themis_secure_message_encrypt([self.privateKey bytes], [self.privateKey length],
+                    [self.publicKey bytes], [self.publicKey length], [message bytes], [message length],
+                    NULL, &wrappedMessageLength);
             break;
 
         case TSMessageModeSignVerify:
             if (!(self.privateKey) || [self.privateKey length] == 0) {
-				if (error) {
-                	*error = SCERROR(TSErrorTypeFail, @"Secure Message failed wraping");
-				}
+                NSLog(@"Error during signing: private key is missing");
+                if (error) {
+                    *error = SCERROR(TSErrorTypeFail, @"Secure Message failed wraping");
+                }
                 return nil;
             }
-            result = (TSErrorType) themis_secure_message_wrap([self.privateKey bytes], [self.privateKey length], NULL, 0,
```

Comment:
`themis_secure_message_wrap` wasn't as public method anywhere? only inside `wrapData` method?

Reply:
yes it was used only here